src/base/quda_bridge.cpp

```diff
@@ -204,7 +204,7 @@ namespace quda_iface
 	
 	initCommsGridQuda(NDIM,grid,get_rank_of_quda_coords,NULL);
 	
-	initQuda(iCudaDevice);
+	initQuda(-1);
```

Comment:
this was the part that resolved the malloc error

Reply:
That's very interesting. This means that on Juwels Booster all four QUDA instances were probably trying to run on a single device because `iCudaDevice` was not set properly. I'm guessing that perhaps another way to get it working would have been to not set `CUDA_VISIBLE_DEVICES=0,1,2,3` in the job script and instead rely on the JSC slurm setting `CUDA_VISIBLE_DEVICES` to a particular value for each MPI task. Having QUDA select its own device (by passing `-1`) is better, however, because it automatically takes into account the node topology AFAIK.